src/adapter/src/coord/appends.rs

```diff
@@ -314,48 +373,14 @@ impl<S: Append + 'static> Coordinator<S> {
     #[tracing::instrument(level = "debug", skip_all, fields(updates = updates.len()))]
     pub(crate) async fn send_builtin_table_updates(&mut self, updates: Vec<BuiltinTableUpdate>) {
         // Most DDL queries cause writes to system tables. Unlike writes to user tables, system
-        // table writes are not batched in a group commit. This is mostly due to the complexity
-        // around checking for conflicting DDL at commit time. There is a possibility that if a user
-        // is executing DDL at a rate faster than 1 query per millisecond, then the global timeline
-        // will unboundedly advance past the system clock. This can cause future queries to block,
-        // but will not affect correctness. Since this rate of DDL is unlikely, we are leaving DDL
-        // related writes out of group commits for now.
-        //
-        // In the future we can add these write to group commit by:
-        //  1. Checking for conflicts at commit time and aborting conflicting DDL.
-        //  2. Delaying modifications to on-disk and in-memory catalog until commit time.
-        let WriteTimestamp {
-            timestamp,
-            advance_to,
-        } = self.get_and_step_local_write_ts().await;
-        let mut appends: HashMap<GlobalId, Vec<(Row, Diff)>> = HashMap::new();
-        for u in updates {
-            appends.entry(u.id).or_default().push((u.row, u.diff));
-        }
-        for (_, updates) in &mut appends {
-            differential_dataflow::consolidation::consolidate(updates);
-        }
-        let appends = appends
-            .into_iter()
-            .map(|(id, updates)| {
-                let updates = updates
-                    .into_iter()
-                    .map(|(row, diff)| Update {
-                        row,
-                        diff,
-                        timestamp,
-                    })
-                    .collect();
-                (id, updates, advance_to)
-            })
-            .collect();
-        self.controller
-            .storage_mut()
-            .append(appends)
-            .expect("invalid updates")
-            .await
-            .expect("One-shot shouldn't fail")
-            .unwrap();
+        // table writes do not wait for a group commit, they explicitly trigger one. There is a
+        // possibility that if a user is executing DDL at a rate faster than 1 query per
+        // millisecond, then the global timeline will unboundedly advance past the system clock.
+        // This can cause future queries to block, but will not affect correctness. Since this
+        // rate of DDL is unlikely, we allow DDL to explicitly trigger group commit.
+        self.pending_writes
+            .extend(updates.into_iter().map(|update| update.into()));
+        self.group_commit().await;
```

Comment:
Oh this is cool. Retains the synchronousness of DDLs because it await on the result.